src/hotspot/share/runtime/os.hpp

```diff
@@ -442,7 +442,7 @@ class os: AllStatic {
 
   // Attempts to reserve the virtual memory at [addr, addr + bytes).
   // Does not overwrite existing mappings.
-  static char*  attempt_reserve_memory_at(char* addr, size_t bytes, bool executable = false);
+  static char*  attempt_reserve_memory_at(char* addr, size_t bytes, bool executable = false, MEMFLAGS flag = mtNone);
```

Comment:
Same here, consider not implementing default value to force the callers to explicitly choose the value?